test/test_surface_temperature.py

```diff
@@ -0,0 +1,191 @@
+        mesh=F.Mesh1D(np.linspace(0, 6.0, 10000)), temperature=500
+    )
+    my_model.define_temperature()
+
+    my_export = F.AverageSurfaceTemperature(
+        filename=filename,
+        surface=F.SurfaceSubdomain1D(id=35, x=1),
+    )
+    my_export.value = 2.0
+    my_export.write(0)
+    my_export.write(1)
+
+    my_export2 = F.AverageSurfaceTemperature(
+        filename=filename,
+        surface=F.SurfaceSubdomain1D(id=1, x=1),
+    )
+    my_export2.value = 3.0
+    my_export2.write(1)
+    my_export2.write(2)
+    my_export2.write(3)
+
+    data = np.genfromtxt(filename, delimiter=",", names=True)
+    file_length = data.size
+    expected_length = 3
+
+    assert file_length == expected_length
+
+
+def test_filename_setter_raises_TypeError():
+    """Test that a TypeError is raised when the filename is not a string"""
+
+    with pytest.raises(TypeError, match="filename must be of type str"):
+        my_model = F.HydrogenTransportProblem(
+            mesh=F.Mesh1D(np.linspace(0, 6.0, 10000)), temperature=500
+        )
+        my_model.define_temperature()
+
+        F.AverageSurfaceTemperature(
+            filename=1,
+            surface=F.SurfaceSubdomain1D(id=1, x=1),
+        )
+
+
+def test_filename_setter_raises_ValueError(tmp_path):
+    """Test that a ValueError is raised when the filename does not end with .csv or .txt"""
+
+    with pytest.raises(ValueError):
+        my_model = F.HydrogenTransportProblem(
+            mesh=F.Mesh1D(np.linspace(0, 6.0, 10000)), temperature=500
+        )
+        my_model.define_temperature()
+
+        F.AverageSurfaceTemperature(
+            filename=os.path.join(tmp_path, "my_export.xdmf"),
+            surface=F.SurfaceSubdomain1D(id=1, x=1),
+        )
+
+
+@pytest.mark.parametrize("value", ["my_export.csv", "my_export.txt"])
+def test_writer(tmp_path, value):
+    """Test that the writes values at each timestep to either a csv or txt file"""
+    my_model = F.HydrogenTransportProblem(
+        mesh=F.Mesh1D(np.linspace(0, 6.0, 10000)), temperature=500
+    )
+    my_model.define_temperature()
+
+    my_export = F.AverageSurfaceTemperature(
+        filename=os.path.join(tmp_path, f"{value}"),
+        surface=F.SurfaceSubdomain1D(id=1, x=0),
+    )
+    my_export.value = 2.0
+
+    for i in range(10):
+        my_export.write(i)
+        file_length = len(np.genfromtxt(my_export.filename, delimiter=","))
+
+        expected_length = i + 2
+
+        assert file_length == expected_length
+
+
+def test_surface_setter_raises_TypeError():
+    """Test that a TypeError is raised when the surface is not a
+    F.SurfaceSubdomain"""
+
+    with pytest.raises(
+        TypeError, match="surface should be an int or F.SurfaceSubdomain"
+    ):
+        my_model = F.HydrogenTransportProblem(
+            mesh=F.Mesh1D(np.linspace(0, 6.0, 10000)), temperature=500
+        )
+        my_model.define_temperature()
+        F.AverageSurfaceTemperature(
+            surface="1",
+        )
```

Comment:
These tests aren't need as it tests the functionality in the parent class `SurfaceQuantity`